network/simple-onchain-discovery/src/discovery.rs

```diff
@@ -0,0 +1,219 @@
+// Copyright (c) The Diem Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use async_trait::async_trait;
+use diem_config::{config::Peer, network_id::NetworkContext};
+use diem_logger::prelude::*;
+use diem_time_service::{TimeService, TimeServiceTrait};
+use diem_types::{account_address::AccountAddress, PeerId};
+use futures::{channel::mpsc::SendError, SinkExt, Stream, StreamExt};
+use network::{
+    connectivity_manager::{ConnectivityRequest, DiscoverySource},
+    logging::NetworkSchema,
+};
+use std::{
+    collections::HashMap,
+    path::Path,
+    pin::Pin,
+    sync::Arc,
+    task::{Context, Poll},
+    time::Duration,
+};
+
+type Result<T> = std::result::Result<T, Error>;
```

Comment:
type Result<T, E = Error> = std::result::Result<T, E>;

Is a little bit more friendly since if you need to you can specify a different error type easily.